### PR TITLE
feat: [PP-10655] Add restriction on container type to echo compatible plates

### PR DESCRIPTION
**Pull request by valtriadi**
Jira Ticket: https://strateos.atlassian.net/browse/PP-10655

Changes included: 

1. Enhanced container type *echo* to have 'acoustic_transfer' capability 
2. Enhanced protocol.acoustic_transfer() to validate input source for the capability

Tests:
Enhanced existing test/protocol_test.py::TestAcousticTransfer for test cases valid & invalid input sources.

### CODE REVIEW

**Review by aaronstrateos — APPROVED** [review]
looks good to me! The scieng team made some updates to acoustic transfer around property propagation that you may have to fold in to this. Once updated with latest master feel free to merge and increment a minor version change. And then follow release instructions for uploading latest appy to gemfury.

https://strateos.atlassian.net/wiki/spaces/PROD/pages/1892909128/2022-09-22+Autoprotocol+Deployment+and+Versioning+Process

**Comment by valtriadi** [comment]
This is odd! After merged from + resolved conflicts with master, @aaronstrateos approval disappeared, thus blocking this PR from merging back to trunk. Have to re-request for review.

**Review by aaronstrateos — APPROVED** [review]

**Pull request merged**
